lib/src/iterable_extensions.dart

```diff
@@ -730,18 +844,11 @@ extension IterableComparableExtension<T extends Comparable<T>> on Iterable<T> {
   ///
   /// The iterable must not be empty.
   T get max {
-    var iterator = this.iterator;
-    if (iterator.moveNext()) {
-      var value = iterator.current;
-      while (iterator.moveNext()) {
-        var newValue = iterator.current;
-        if (value.compareTo(newValue) < 0) {
-          value = newValue;
-        }
-      }
-      return value;
+    final result = maxOrNull;
+    if (result == null) {
+      throw StateError('No element');
     }
-    throw StateError('No element');
```

Comment:
Yes, this should be safe because a nullable type cannot possibly implement `Comparable<T>` for any `T`.